tests/WindowsUpdateCheckTest.php

```diff
@@ -0,0 +1,32 @@
+<?php declare(strict_types=1);
+
+namespace Keyman\Site\com\keyman\api\tests;
+
+require_once(__DIR__ . '/../tools/base.inc.php');
+require_once(__DIR__ . '/../script/windows/14.0/update/WindowsUpdateCheck.php');
+require_once(__DIR__ . '/ApiTestCase.php');
+require_once(__DIR__ . '/TestUtils.inc.php');
+
+use Keyman\Site\com\keyman\api\tests\ApiTestCase;
+
+final class WindowsUpdateCheckTest extends ApiTestCase
+{
+  private const SchemaFilename = "/windows-update/14.0/windows-update.json";
+
+  public function testSimpleResultValidatesAgainstSchema(): void
+  {
+    $schema = TestUtils::LoadJSONSchema(self::SchemaFilename);
+
+    $u = new \Keyman\Site\com\keyman\api\WindowsUpdateCheck();
+    $json = $u->execute('alpha', '14.0.100', ['khmer_angkor'=>'1.0.2', 'foo'=>'0.1']);
```

Comment:
From this, I take it that this parallels an in-app "check for updates" menu item?  Said menu item then checks to update both Keyman for Windows and any resources that are installed?

Reply:
At present, this API endpoint is used by the setup bootstrap program, which looks for latest Keyman Desktop as well as keyboards specified by parameter. While the Keyman Desktop online update check will move to use this as well, I have not yet checked the code around that.